controller/backup_controller.go

```diff
@@ -659,7 +659,9 @@ func (bc *BackupController) checkMonitor(backup *longhorn.Backup, volume *longho
 		return nil, err
 	}
 
-	if engine.Status.CurrentState != longhorn.InstanceStateRunning {
+	if engine.Status.CurrentState != longhorn.InstanceStateRunning ||
+		engine.Spec.DesireState != longhorn.InstanceStateRunning ||
+		volume.Status.State != longhorn.VolumeStateAttached {
```

Comment:
Perhaps split this into two separate checks, so that the error returned can accurately reflect that it is engine state or volume state that is the problem.

Reply:
Agreed, but in this case, it could be optional because all conditions here are mandatory to just ensure the volume is really attached.